### PR TITLE
JCache Split-Brain code coverage improvement

### PR DESCRIPTION
**Pull request by serkan-ozal**

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/cache/CacheLoadAllTest.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        // Verify that backup of load-all works
+        for (Map.Entry<String, String> entry : entries.entrySet()) {
+            String key = entry.getKey();
+            String expectedValue = entries.get(key);
+            Data keyData = serializationService.toData(key);
+            int keyPartitionId = partitionService.getPartitionId(keyData);
+            for (int i = 0; i < INSTANCE_COUNT; i++) {
+                Node n = getNode(hazelcastInstances[i]);
+                ICacheService cacheService = n.getNodeEngine().getService(ICacheService.SERVICE_NAME);
+                ICacheRecordStore recordStore = cacheService.getRecordStore("/hz/" + cacheName, keyPartitionId);
```
Comment (bilalyasar) [inline hazelcast/src/test/java/com/hazelcast/cache/CacheLoadAllTest.java]: here , do you read backup? i couldn't get the idea.

Comment (serkan-ozal) [inline hazelcast/src/test/java/com/hazelcast/cache/CacheLoadAllTest.java]: yep, I am checking backup to verify that `LoadAllOperation` is mirrored to backup also

File: hazelcast/src/test/java/com/hazelcast/cache/CacheManagementTest.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    }
+
+    @Test
+    public void cacheMXBeanShouldBeRegisteredWhenManagementIsEnabled() {
+        doCacheManagement(true, false);
+    }
+
+    @Test
+    public void cacheMXBeanShouldNotBeRegisteredWhenManagementIsEnabled() {
+        doCacheManagement(false, false);
```
Comment (bilalyasar) [inline hazelcast/src/test/java/com/hazelcast/cache/CacheManagementTest.java]: you also have, `cacheMXBeanShouldBeRegisteredWhenManagementIsEnabled `?.. isn't it contradiction?

File: hazelcast/src/test/java/com/hazelcast/cache/CacheManagementTest.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    }
+
+    @Test
+    public void cacheMXBeanShouldBeRegisteredWhenStatisticsIsEnabled() {
+        doCacheManagement(true, true);
+    }
+
+    @Test
+    public void cacheMXBeanShouldNotBeRegisteredWhenStatisticsIsEnabled() {
+        doCacheManagement(false, true);
```
Comment (bilalyasar) [inline hazelcast/src/test/java/com/hazelcast/cache/CacheManagementTest.java]: you also have, `cacheMXBeanShouldBeRegisteredWhenStatisticsIsEnabled ` ? isn't it contradiction?

Comment (serkan-ozal) [inline hazelcast/src/test/java/com/hazelcast/cache/CacheManagementTest.java]: ah, yes. bad naming

File: hazelcast/src/test/java/com/hazelcast/cache/config/CacheConfigReadOnlyTest.java [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+import com.hazelcast.config.CacheSimpleConfigReadOnly;
+import com.hazelcast.config.InMemoryFormat;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category(QuickTest.class)
```
Comment (bilalyasar) [inline hazelcast/src/test/java/com/hazelcast/cache/config/CacheConfigReadOnlyTest.java]: why serial?

File: hazelcast/src/test/java/com/hazelcast/cache/config/CacheSimpleConfigReadOnlyTest.java [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+import com.hazelcast.config.CacheSimpleConfigReadOnly;
+import com.hazelcast.config.InMemoryFormat;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category(QuickTest.class)
```
Comment (bilalyasar) [inline hazelcast/src/test/java/com/hazelcast/cache/config/CacheSimpleConfigReadOnlyTest.java]: why serial?

**Comment by bilalyasar** [comment]
minor comments, looks good to me :green_apple:

File: hazelcast/src/test/java/com/hazelcast/cache/config/CacheConfigReadOnlyTest.java (hunk 4 above)
Comment (serkan-ozal) [inline hazelcast/src/test/java/com/hazelcast/cache/config/CacheConfigReadOnlyTest.java]: they are registering MXBeans so I didn't want that they effect eachother

**Comment by serkan-ozal** [comment]
thanks @bilalyasar for your review

File: hazelcast/src/test/java/com/hazelcast/cache/CacheLoadAllTest.java (hunk 1 above)
Comment (bilalyasar) [inline hazelcast/src/test/java/com/hazelcast/cache/CacheLoadAllTest.java]: so, `cacheService.getRecordStore("/hz/" + cacheName, keyPartitionId);` returns backup?

Comment (serkan-ozal) [inline hazelcast/src/test/java/com/hazelcast/cache/CacheLoadAllTest.java]: ıt can be from owner or backup. But when we get from all partitionS on all nodes, we sure that we have checked all owned and backup partitions

Comment (bilalyasar) [inline hazelcast/src/test/java/com/hazelcast/cache/CacheLoadAllTest.java]: if cacheService has backup of the data, it uses backup data, it doesn't send request to owner. am i right?

Comment (serkan-ozal) [inline hazelcast/src/test/java/com/hazelcast/cache/CacheLoadAllTest.java]: right, we get them from `ICacheRecordStore` itself directly without any operation.

**Pull request merged**